### ISSUE TITLE
SyntaxError: Functions cannot be declared in a nested block in strict mode

### ISSUE THREAD

**Issue by JCCR** [gildas-lormeau/zip.js#168]
I'm trying to use this library on a project running with the headless browser PhantomJS 2.1 (http://phantomjs.org/)

But simply including this library into my project causes this error to appear when loading the javascript:
`SyntaxError: Functions cannot be declared in a nested block in strict mode`

The problem is in this line:
https://github.com/gildas-lormeau/zip.js/blob/master/WebContent/zip-fs.js#L163

My fix is to replace lines 158 to 177 with this snippet:

```js
function readEntries(directoryReader, callback, entries, onerror) {
    directoryReader.readEntries(function(temporaryEntries) {
        if (!temporaryEntries.length)
            callback(entries);
        else {
            entries = entries.concat(temporaryEntries);
            readEntries(directoryReader, callback, entries, onerror);
        }
    }, onerror);
}

function addFileEntry(zipEntry, fileEntry, onend, onerror) {
    function getChildren(fileEntry, callback) {
        var entries = [];
        if (fileEntry.isDirectory) {
            var directoryReader = fileEntry.createReader();
            readEntries(directoryReader, callback, entries, onerror);
        }
        if (fileEntry.isFile)
            callback(entries);
    }
```

**Comment by Rob--W** [comment]
An alternative would be

```javascript
(function readEntries() {
 ....
})();
// Removed readEntries();
```

Pull requests are welcome, thanks for the report.

**Comment by Rob--W** [comment]
Fixed via #170.
